test/Test/Database/LSMTree/Internal/Snapshot.hs

```diff
@@ -0,0 +1,262 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Test.Database.LSMTree.Internal.Snapshot (tests) where
+
+import           Codec.CBOR.Decoding
+import           Codec.CBOR.Encoding
+import           Codec.CBOR.FlatTerm
+import           Codec.CBOR.Read
+import           Codec.CBOR.Write
+import qualified Data.ByteString.Lazy as BSL
+import           Data.Proxy
+import           Data.Text (Text)
+import qualified Data.Text as Text
+import qualified Data.Vector as V
+import           Database.LSMTree.Internal.Config
+import           Database.LSMTree.Internal.Entry
+import           Database.LSMTree.Internal.MergeSchedule
+import           Database.LSMTree.Internal.RunNumber
+import           Database.LSMTree.Internal.Snapshot
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+
+-- TODO: we should add golden tests for the CBOR encoders. This should prevent
+-- accidental breakage in the format.
```

Comment:
> We can make sure we read existing snapshots correctly (especially after minor version updates) ...

Yes, it would be useful to test that backwards compatibility works.

> ... and also ensure we produce exactly the expected snapshot files. However, for the latter one, we have to be careful when making changes to the merging algorithm. We should probably specify the test inputs as exact table structure (instead of creating them by inserting values, which could lead to different level structures in the future).

Yeah this is slightly more tricky. We can look into it once we add the tests